source/MaterialXTest/Material.cpp

```diff
@@ -27,6 +27,7 @@ TEST_CASE("Material", "[material]")
 
     // Create an inherited shader nodedef.
     mx::NodeDefPtr anisoSrf = doc->addNodeDef("ND_anisoSrf", "surfaceshader", "anisoSrf");
+    anisoSrf->removeOutput("out");
```

Comment:
This is currently necessary as creating a NodeDef adds a child output called "out" to it, so in this case both the base class simpleSrf and derived class anisoSrf both have an output called "out". This breaks things as a node with more than one output is treated as a multioutput so any attempts to find the nodedef fail as "multioutput" gets compared to "surfaceshader". We can certainly discuss alternate solution for this, as Bernard and I aren't fans of this solution.

Reply:
Would it be possible to let nodedef inheritance only be allowed to set at nodedef creation time? That is, let the inherits attribute be set in the Document::addNodeDef call. That way the inheritance information is available when creating the inputs and outputs, and the "out" output can be optionally created.

I suppose the "out" output should never be created if inheritance is used, since we can assume the base nodedef has either "out" or multiple other outputs already defined.

Reply:
A few comments:
- So you're suggesting that we kill the setInheritsFrom method and instead add a parameter toaddNodeDef that specifies the parent? If a parent is specified don't create "out"? That might work.
- An alternative is to pass "" as the type to specify that it shouldn't have an "out" created, but maybe your solution is better.
- I still don't fully get how inheritance in MaterialX works. Can I have two inputs with the same name in the base and derived classes to override the value of an input, or are duplicate inputs simply not allowed?

Reply:
As we discussed offline a better solution is to update `getActiveOutputs` and `getActiveInputs`, so they never return ports with the same name from inherited nodedefs.